packages/sanity/src/core/form/inputs/files/ImageInput/ImageActionsMenu.tsx

```diff
@@ -15,12 +13,22 @@ interface ImageActionsMenuProps {
   children: ReactNode
   onEdit: MouseEventHandler<HTMLButtonElement>
   setHotspotButtonElement: (element: HTMLButtonElement | null) => void
-  setMenuButtonElement: (element: HTMLButtonElement | null) => void
   showEdit: boolean
+  isMenuOpen: boolean
+  onMenuOpen: (flag: boolean) => void
 }
 
 export function ImageActionsMenu(props: ImageActionsMenuProps) {
-  const {onEdit, children, showEdit, setHotspotButtonElement, setMenuButtonElement} = props
+  const {onEdit, children, showEdit, setHotspotButtonElement, onMenuOpen, isMenuOpen} = props
+
+  const [menuElement, setMenuRef] = useState<HTMLDivElement | null>(null)
+
+  const handleClick = React.useCallback(() => onMenuOpen(true), [onMenuOpen])
```

Comment:
Nitpick (feel free to ignore): I prefer to not use `React.useCallback` but instead `import {useCallback} from 'react'`